photutils/psf/photometry.py

```diff
@@ -171,15 +153,63 @@ def aperture_radius(self, value):
                              'number, received aperture_radius = {}'
                              .format(value))
 
+    def get_residual_image(self):
+        """
+        Returns an image that is the result of the subtraction between
+        the original image and the fitted sources.
+
+        Returns
+        -------
+        residual_image : 2D array-like, `~astropy.io.fits.ImageHDU`, `~astropy.io.fits.HDUList`
+        """
+
+        return self._residual_image
+
     def __call__(self, image, positions=None):
         """
-        Performs PSF photometry using the DAOPHOT algorithm. See
-        `do_photometry` for more details including the `__call__`
-        signature.
+        Performs PSF photometry. See `do_photometry` for more details
```

Comment:
For anyone reading this: we ended up hashing this out via google hangouts: the plan is to go back to the `__call__` is the same as `do_photometry` situation by adding another method that does only what the current `do_photometry` does, and moving the current `__call__` to `do_photometry`